wordpress-mobile/WordPress-iOS
# Media Picker: Fix the no results message position

Fixes https://github.com/wordpress-mobile/WordPress-iOS/issues/18840, https://github.com/wordpress-mobile/WordPress-iOS/issues/19514
WPMediaPicker: https://github.com/wordpress-mobile/MediaPicker-iOS/pull/399 

## Description
This PR is the union of the changes made on the opened PR to the `WPMediaPicker-iOS` pod and a minor fix on the `WordPress-iOS` project side.

### Issue on WPMediaPicker-iOS:
It was observed after the user started searching for some media and returned a no-result media screen, **_the message presented was not showing correctly when the keyboard popped up_**.

#### Reason:
When the keyboard pops up, the code changes the `collectionView` `contentInset` and, in sequence, changes the frame size and position of the empty view based on the new `contentInset` values.

![bug_before_current_wpmediapicker](https://user-images.githubusercontent.com/20734091/199581694-afb6ae7e-3cc0-4474-971a-99de5639640a.gif)

#### What changed on the WPMediaPicker pod?
Now we have a UIView container to receive the `emptyView` from the delegate. This container is attached in the `collectionView` using constraints to anchor. When the keyboard pops up, we only need to change the bottom constraint to update the container size, getting the expected result. Also, all views on the `WPMediaPickerViewController` are now attached to the view using the constraint auto layout.

![fix_after_wpmediapicker_beta](https://user-images.githubusercontent.com/20734091/199581771-51375fa9-5101-4954-bbe5-1164ceedcf53.gif)

#### What changes on the WP-iOS project?
Even using the new beta version of the Pod, in landscape mode, the message is showing behind the keyboard. This message belongs to `NoResultsViewController`, so a change was made to update the top constraint for landscape mode.

![Simulator Screen Recording - iPhone 13 - 2022-11-03 at 22 47 39](https://user-images.githubusercontent.com/20734091/199867635-9822ae11-c3d1-4517-a197-d15794564cff.gif)

## Testing instructions
1. Run the app
2. Switch to a site that has media uploaded
3. Open Blog Post and add an `IMAGE` block
4. Make sure that media are displayed correctly on the list
5. Opens the Keyboard
6. Search to get no results
7. Make sure that the message `No media matching your search` is displayed correctly in Portrait and Landscape modes, with and without the keyboard on the screen.

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
Added test on `NoResultsViewControllerTests`

PR submission checklist:
